stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/usecase/QueryMembersTest.kt

```diff
@@ -0,0 +1,163 @@
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.utils.FilterObject
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.livedata.ChatDomainImpl
+import io.getstream.chat.android.livedata.randomUser
+import io.getstream.chat.android.livedata.repository.RepositoryFacade
+import io.getstream.chat.android.test.TestCall
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.amshove.kluent.When
+import org.amshove.kluent.`should be equal to`
+import org.amshove.kluent.any
+import org.amshove.kluent.calling
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+internal class QueryMembersTest {
+
+    private lateinit var mockDomain: ChatDomainImpl
+    private lateinit var mockClient: ChatClient
+    private lateinit var mockRepo: RepositoryFacade
+    private lateinit var queryMembers: QueryMembers
+
+    private val channelType = "test"
+    private val channelId = "channel"
+    private val channel = Channel(cid = "$channelType:$channelId", type = channelType, id = channelId)
+    private val filter = FilterObject()
+    private val sort = QuerySort.desc(Member::createdAt)
+    private var onlineUsers = listOf(
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser()
+    )
+    private var offlineUsers = listOf(
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser(),
+        randomUser()
+    )
+    private val onlineMembers = onlineUsers.map(::Member)
+    private val offlineMembers = offlineUsers.map(::Member)
+
+    @BeforeEach
+    fun setup() = runBlocking {
+        mockClient = mock { client ->
+            on(client.queryMembers(any(), any(), any(), any(), any(), any(), any())) doReturn TestCall(
+                Result(
+                    onlineMembers
+                )
+            )
+        }
+
+        mockRepo = mock { repo ->
+            on(repo.selectMembersForChannel(channel.cid)) doReturn offlineMembers
+        }
+
+        mockDomain = mock { domain ->
+            on(domain.client) doReturn mockClient
+            on(domain.repos) doReturn mockRepo
+            on(domain.scope) doReturn TestCoroutineScope()
+            on(domain.offlineEnabled) doReturn true
+        }
+
+        queryMembers = QueryMembers(mockDomain)
+    }
+
+    @Test
+    fun `Given default arguments, and ONLINE, SHOULD make a remote call via client`() = runBlockingTest {
+        When calling mockDomain.isOffline() doReturn false
```

Comment:
What about not using kluent anymore as we discussed in the last catch-up?

Reply:
Sure, can update it.